panic: runtime error: invalid memory address or nil pointer dereference

Hello,

First of all, congrats on the amazing project! It was running perfectly on my homelab for quite long.

Not even sure if this is a coding-related issue or something else.

The pifrost it's the only application crashing on my homelab. It starts and run for like 3-5 minutes and show this error message:

```
time="2023-03-09T16:19:11Z" level=info msg="Creating DNS Provider" host=192.168.100.12 insecure=true
time="2023-03-09T16:19:11Z" level=info msg="Attempting to reach pi-hole..."
time="2023-03-09T16:19:11Z" level=info msg=Connected.
time="2023-03-09T16:19:11Z" level=info msg="Starting ingress watcher..."
time="2023-03-09T16:19:11Z" level=info msg="Externalizing all ingress objects"
time="2023-03-09T16:19:11Z" level=info msg="Starting service watcher..."
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x120e3f7]

goroutine 33 [running]:
github.com/tolson-vkn/pifrost/watcher.watcherIngress(0xc0003f7b60, %
```

I tested with both v0.1.0 and v0.1.1 images and also increased the number of replicas to spread across the nodes. All the replicas fail with the same message.

Appreciate your insights!

--- a/watcher/watcher.go
+++ b/watcher/watcher.go
@@ -72,9 +72,7 @@ func watcherIngress(dnsProvider *provider.PiHoleRequest, client *kubernetes.Clie
 			if len(ingressEIP) == 0 {
 				ingress, err = pollIngress(ingress, client)
 				if err != nil {
-					logrus.WithFields(logrus.Fields{
-						"ingress": ingress.ObjectMeta.Name,
-					}).Warnf("Watch error: %s", err)
+					logrus.Warnf("Watch error: %s", err)
 					continue
 				}
 
@@ -130,9 +128,7 @@ func watcherIngress(dnsProvider *provider.PiHoleRequest, client *kubernetes.Clie
 			if len(ingressEIP) == 0 {
 				ingress, err = pollIngress(ingress, client)
 				if err != nil {
-					logrus.WithFields(logrus.Fields{
-						"ingress": ingress.ObjectMeta.Name,
-					}).Warnf("Watch error: %s", err)
+					logrus.Warnf("Watch error: %s", err)
 					continue
 				}
 
@@ -217,9 +213,7 @@ func watcherService(dnsProvider *provider.PiHoleRequest, client *kubernetes.Clie
 			if ok {
 				svc, err = pollService(svc, client)
 				if err != nil {
-					logrus.WithFields(logrus.Fields{
-						"service": svc.ObjectMeta.Name,
-					}).Warnf("Watch error: %s", err)
+					logrus.Warnf("Watch error: %s", err)
 					continue
 				}
 				if svc.Spec.Type == "LoadBalancer" {

